packages/next-auth/src/next/index.ts

```diff
@@ -141,12 +146,18 @@ export async function unstable_getServerSession(
     options = args[2]
   }
 
-  options.secret = options.secret ?? process.env.NEXTAUTH_SECRET
+  options.secret ??= process.env.NEXTAUTH_SECRET
+  options.trustHost ??= !!(process.env.AUTH_TRUST_HOST ?? process.env.VERCEL)
 
   const session = await NextAuthHandler<Session | {} | string>({
     options,
     req: {
-      host: detectHost(req.headers["x-forwarded-host"]),
+      host: detectHost(
+        options.trustHost,
+        req.headers["x-forwarded-host"],
+        process.env.NEXTAUTH_URL ??
+          (process.env.NODE_ENV !== "production" && "http://localhost:3000")
```

Comment:
@balazsorban44 

I am not sure if hardcoding the port makes sence here. This might break the experience since Next.js might choose another port if:

* the `-p` command line switch is used (e.g.: `next dev -p 4000`)
* or port `3000` is already in use. 

Afaik Node.js & the Edge Env add an Host header anyway, when using the http protocol. Is there any reason, we are not using this `req.host` or `req.headers.get('host')` as a fallback? I have just tested this using `next dev` and it seems to work in edge and node.js envs.

### Docs

- [`http.host`](https://nodejs.org/api/http.html#requesthost)